Overhaul toolbar on Mac app

AngeloStavrow opened the pull request:
Closes #128.

This PR doesn't yet include toolbar UI/UX (if any) to support moving posts between blogs (todo in #134). Demo of key toolbar functionality:

![wf-mac-toolbar-features](https://user-images.githubusercontent.com/387655/100486060-7d647b00-30d0-11eb-98ab-0fff761bd74e.gif)

This PR also adds a few menubar entries and keyboard shortcuts for some functions:

- File &rarr; New Local Draft (CMD + N)
- File &rarr; Reload From Server (CMD + R)
- View &rarr; Toggle Sidebar (CMD + OPT + S)

This PR also fixes #127.

Demo of some menu commands and the Edit &rarr; Delete fix:

![wf-mac-menubar-features](https://user-images.githubusercontent.com/387655/100486312-58bcd300-30d1-11eb-8619-21b21427f3b1.gif)

Some known issues remain and I'll open GitHub issues for them when this is merged in, as they're beyond the scope of this PR:

- [ ] The blog title and number of posts in the toolbar doesn't update unless you click on "All Posts" in the toolbar once after launch.
- [ ] The app sometimes crashes doing the above.
- [ ] The reload-from-server button occasionally disappears and (usually) animates back when a new blog is selected in the sidebar.

thebaer reviewed (changes requested):
Overall looks good after reading through! 

In addition to these wording changes, I think we should put the Publish button in the right-most position on the toolbar, since that's the primary action users will be taking on a post.

Also, what would you think of removing the Refresh button from the toolbar? To me, it looks a bit odd to the left of the title (particularly with the sidebar shown). And I'm not sure it'll be used frequently enough to warrant space on the toolbar; I'd say if we can keep visual elements to a minimum (in the name of being distraction-free), that'd be best.

AngeloStavrow commented on Shared/WriteFreely_MultiPlatformApp.swift:
```diff
@@ -22,6 +23,35 @@ struct WriteFreely_MultiPlatformApp: App {
         }
+        .commands {
+            CommandGroup(replacing: .newItem, addition: {
+                Button("New Local Draft") {
```
Sounds good!

AngeloStavrow replied:
Fixed in 24897aa — as shown in this screenshot:

<img width="613" alt="menu_new-post" src="https://user-images.githubusercontent.com/387655/100754810-7df35f00-33b9-11eb-8624-355b70db05c8.png">

AngeloStavrow commented on Shared/WriteFreely_MultiPlatformApp.swift:
```diff
@@ -22,6 +23,35 @@ struct WriteFreely_MultiPlatformApp: App {
+                .keyboardShortcut("n", modifiers: [.command])
+            })
+            CommandGroup(after: .newItem) {
+                Button("Reload From Server") {
```
I'm all for better wording! Looking at some terms used by other apps:

- "Reload Page" (Mac App Store)
- "Update With Things Cloud" (Things)
- "Sync Now" (OmniFocus)
- "Reload" (GitKraken)
- "Refresh Calendars" (Calendar)

@thebaer Happy to go with "Refresh"/"Refresh Posts", but maybe something like "Update From Server" (or "Update From WriteFreely" / "Update from Write.as" as appropriate) might be clearer? My instinct is that the messaging should reinforce that this is a one-way server&rarr;client sync. What do you think?

thebaer replied:
Agreed we should definitely steer clear of using "Sync," but I think "Refresh" or "Reload" captures that one-way relationship well enough -- it's basically using a browser analogy.

My biggest issue is with words like "server" (which isn't a layman's term) and "cloud" (which is abstract / remote from the user). Purely from the user's perspective, they click _reload_ / _refresh_ and all of their posts and things are updated -- that's it. They only need to understand that clicking the button == updating the screen, just like a web page or their social media feed. So I think "Refresh" or "Reload", potentially followed by a noun, would be a nice, concise way to convey what that button does.

AngeloStavrow replied:
How about "Fetch Changes"?

(As proposed in Slack)

AngeloStavrow commented:
> In addition to these wording changes, I think we should put the Publish button in the right-most position on the toolbar, since that's the primary action users will be taking on a post.

Fixed in 43088fd, see screenshot:

<img width="326" alt="Screen Shot 2020-12-01 at 10 01 22" src="https://user-images.githubusercontent.com/387655/100757542-7e412980-33bc-11eb-974a-689f0456012e.png">

> Also, what would you think of removing the Refresh button from the toolbar? To me, it looks a bit odd to the left of the title (particularly with the sidebar shown). And I'm not sure it'll be used frequently enough to warrant space on the toolbar; I'd say if we can keep visual elements to a minimum (in the name of being distraction-free), that'd be best.

Fixed in 7902bff and c1be327; also took a second to make "Cancel" the default action in the delete-confirmation alert, since making sure that works (via the Edit &rarr; Delete menu item) is part of my testing. This also means I can get rid of the hack-y fix I was using to make the buttons equidistant when the sidebar was collapsed, which makes for a cleaner animation. 🙂 

![sidebar-toggle](https://user-images.githubusercontent.com/387655/100758341-661dda00-33bd-11eb-97bd-2b481878d019.gif)

thebaer commented:
Toolbar updates look great!

thebaer reviewed (approved):
Looks great, thanks @AngeloStavrow!
